drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -83,7 +84,17 @@
 kinsol.v = v;
 
 if (options.use_mex && model.mex_model_ptr~=0 && isnumeric(q))
-  doKinematicsmex(model.mex_model_ptr,q,options.compute_gradients,v,options.compute_JdotV);
+  if isempty(options.kinematics_cache_ptr_to_use)
+    if options.compute_gradients
+      kinsol.mex_ptr = model.default_kinematics_cache_ptr_with_gradients;
```

Comment:
The way I see it, I'm not breaking any of the existing Matlab code; the previous behavior of having the order of mex doKinematics calls matter is retained by default. However, you do now have the option of having the order of mex doKinematics calls *not* matter through the kinematics_cache_ptr_to_use option. So I'm adding a feature instead of breaking an existing contract. In the long run, it would be good to get rid of the default caches, but I didn't want to figure out what a good place to store the kinematics cache pointer is for every current use case of Matlab doKinematics.